### PR TITLE
Travis ci without docker

### PR DESCRIPTION
**Pull request by mjhenkes**
### Summary
Testing to see if we could remove docker from travis-ci and if that would give us any performance gains.

Closes #

### Deployment Link

https://terra-core-deployed-pr-#.herokuapp.com/

### Additional Details
The initial terra docs repo PR:
Screen Shot 2020-07-12 at 10 49 14 AM

This PR:
Screen Shot 2020-07-12 at 10 48 22 AM

This PR: grouping like times:
Screen Shot 2020-07-13 at 10 32 47 AM

Compare the `ran for` times, total time indicates waiting in executors (i think)

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,106 +7,157 @@ branches:
...
-    - stage: compile - default theme
-      script: travis_wait docker-compose run test-ci npm run compile:prod -- --env.disableAggregateThemes
+    - stage: danger lint jest and compile
+      script:
+        # clear the cache of any old build artifacts
+        - rm -rf ./travis-build
+        - npm run danger
+        - npm run lint
+        - npm run jest
+        # These can't be built in parallel because the cache is a single object and you can't add to it in parallel
```
Comment (ryanthemanuel) [inline .travis.yml:23]: Can we run all 3 of these in parallel with danger lint and jest?

File: wdio.conf.js [hunk 2]
```diff
@@ -8,4 +8,12 @@ if (process.env.npm_package_name !== 'terra-core') {
   wdioConfig.specs = [path.join(__dirname, 'packages', process.env.npm_package_name, 'tests', 'wdio', '**', '*-spec.js')];
 }
 
+const ci = process.env.CI;
+
+if (ci) {
+  wdioConfig.host = 'localhost';
+}
+
+// console.log(wdioConfig);
```
Comment (ryanthemanuel) [inline wdio.conf.js:17]: Remove commented console log.

File: .travis.yml (hunk 1 above)
Comment (ryanthemanuel) [reply .travis.yml:23]: Never mind. I think I understand your comment now.

Comment (mjhenkes) [reply .travis.yml:23]: Yeah at one point i had all three compile jobs running in parallel but only the last to complete would be cached.

We could get a couple of minutes back by deprecating terra-markdown and terra-doc-template (the only remaining   components that use theme aggregation)

File: wdio.conf.js [hunk 3]
```diff
@@ -8,4 +8,10 @@ if (process.env.npm_package_name !== 'terra-core') {
   wdioConfig.specs = [path.join(__dirname, 'packages', process.env.npm_package_name, 'tests', 'wdio', '**', '*-spec.js')];
 }
 
+const ci = process.env.CI;
+
+if (ci) {
+  wdioConfig.host = 'localhost';
```
Comment (mjhenkes) [inline wdio.conf.js:14]: This line won't work in the docker dev env.... if you use it and i think it's only me... comment this out to run tests

File: package.json [hunk 4]
```diff
@@ -47,7 +47,7 @@
   },
   "scripts": {
     "bootstrap:hoist": "lerna bootstrap --hoist --nohoist=one-cerner-style-icons",
-    "clean": "lerna clean --yes && rm -rf node_modules",
+    "clean": "lerna clean --yes && rm -rf node_modules || true",
```
Comment (mjhenkes) [inline package.json:50]: this was added to allow this check to succeed even if lerna isn't installed.... no need to clean then right?

File: wdio.conf.js (hunk 3 above)
Comment (mjhenkes) [reply wdio.conf.js:14]: i changed this to look for a different env `travis` this will keep docker dev env working

**Pull request merged**